### ISSUE TITLE
pandas : ep_2, эзотерика

### ISSUE THREAD

**Issue by avasyukov** [ermakov-ad/python_labs#5]
Кажется, можно всё-таки без вот этой слегка эзотерической формулы:
https://github.com/ermakov-ad/python_labs/blob/42f2de6f59e379117e9ae4dba036337e7f4d6735/ep_2.py#L11

Если она прямо неизбежно нужна, то к ней нужны явные комментарии в коде, что это, почему оно именно такое, что значат волшебные константы.

P.S. Ну и в целом `.apply()` лучше избегать. Он очень медленный.

**Comment by ermakov-ad** [comment]
да, я помню ту часть лекции, где вы говорили не использовать apply без крайней необходимости. Я пытался обойтись без неё, но у меня не вышло реализовать счётчик количества строк в общем виде. В одном столбце исходной таблицы значения пронумерованы, тогда при использовании  data_sum.iloc[:, 0] (после применения группировки по компаниям), в этом столбике будут значения суммы от 1 до n. Эта формула - обратная к ней, т.е. f(sum) = n, что требовалось в задаче

**Comment by avasyukov** [comment]
О как. Понятно. Думаю, прекрасно подойдёт после `groupby` применить `count` (по аналогии с `sum`).

**Comment by ermakov-ad** [comment]
Переписал этот момент. Изначально думал, что по времени будет рациональнее применить apply, чем дважды преобразовывать таблицу (sum всё равно нужен для ответа на другие вопросы задачи), поэтому реализация была такой
